share/p2p/discovery/discovery.go

```diff
@@ -181,16 +175,13 @@ func (d *Discovery) Advertise(ctx context.Context) {
 
 			// we don't want retry indefinitely in busy loop
 			// internal discovery mechanism may need some time before attempts
-			errTimer := time.NewTimer(retryTimeout)
 			select {
-			case <-errTimer.C:
-				errTimer.Stop()
+			case <-time.After(d.params.AdvertiseRetryTimeout):
```

Comment:
Generally, `time.After` should be avoided as it's leaky. 
(It won't be after 1.23 tho)